server/lib/nl/utils.py

```diff
@@ -566,7 +566,9 @@ def place_detection_with_heuristics(query_fn, query: str) -> List[str]:
   # First check in special places. If they are found, add those first.
   places_found = []
   for special_place in constants.OVERRIDE_FOR_NER:
-    if special_place in query.lower():
+    # Matching <space>special_place<space> because otherwise "asia" could
+    # also match "asian" which is undesirable.
+    if f" {special_place} " in f" {query.lower()} ":
```

Comment:
why not do equality check then?

Reply:
Not sure I follow. 
Equality check would only work if we were guaranteed that we are always matching one word (e.g. "asia") with another word but we can often be matching more than one word (e.g. "sf bay area") against a full query string.

Reply:
ah ok